### PR TITLE
Adding file remotes and sync command

### PR DESCRIPTION
**Pull request by daviddavis**

### CODE REVIEW

File: pulpcore/cli/file_cli.py [hunk 1]
```diff
@@ -30,29 +32,51 @@ def list(ctx):
     ctx.obj.output_result(result)
 
 
+def _find_remote(ctx, name):
+    search_result = ctx.obj.call("remotes_file_file_list", parameters={"name": name, "limit": 1})
+    if search_result["count"] != 1:
+        raise Exception(f"Remote '{name}' not found.")
+    remote = search_result["results"][0]
+    return remote["pulp_href"]
```
Comment (mdellweg) [inline pulpcore/cli/file_cli.py:40]: It's maybe more useful if we return the complete remote here.

File: tests/scripts/test_file_sync.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+. "$(dirname "$(realpath $0)")/config.sh"
+. "$(dirname "$(realpath $0)")/constants.sh"
+
+pulp_cli file remote create --name "cli_test_file_remote" --url "$FILE_REMOTE_URL"
+pulp_cli file repository create --name "cli_test_file_repository" --remote "cli_test_file_remote"
+
+pulp_cli file repository sync --name "cli_test_file_repository"
```
Comment (mdellweg) [inline tests/scripts/test_file_sync.sh:9]:
```suggestion
pulp_cli file repository sync --name "cli_test_file_repository"
pulp_cli file repository sync --name "cli_test_file_repository" --remote "cli_test_file_remote"
```

**Comment by mdellweg** [comment]
This looks great.
As you mentioned in the TODO, we need to find a better way to organize the code and decide how to register the commands.

**Review by mdellweg — CHANGES_REQUESTED** [review]

File: pulpcore/cli/file_cli.py [hunk 3]
```diff
@@ -65,3 +89,26 @@ def destroy(ctx, name):
...
+    repository_href = search_result["results"][0]["pulp_href"]
+
+    if remote:
+        remote_href = _find_remote(ctx, remote)["pulp_href"]
+        body["remote"] = remote_href
+
+    ctx.obj.call(ctx.obj.sync_id, parameters={ctx.obj.href_key: repository_href}, body=body)
+
+
+# TODO: figure out how we want to load/organize these commands
```
[the remark this answers is not in the document]
Comment (daviddavis) [reply pulpcore/cli/file_cli.py:113]: So you're proposing that in remote.py I do this?

```
from pulpcore.file_cli import file

@file.group()
def remote(ctx, remote_type):
    #...
```

Then how would I load the commands from remote.py?

[the remark this answers is not in the document]
Comment (mdellweg) [reply pulpcore/cli/file_cli.py:113]: I'm not sure i follow. Are you saying subcommands from submodules do not appear until the submidule is loaded by other means?

**Review by mdellweg — APPROVED** [review]

**Pull request merged**